packages/dd-trace/test/appsec/iast/analyzers/nosql-injection-mongodb-analyzer.mquery.plugin.spec.js

```diff
@@ -0,0 +1,176 @@
+      const satisfiesNodeVersionForMongo6 =
+        (semver.satisfies(process.version, '>=16.20.1') && semver.satisfies(mongodb.version(), '>=6'))
+
+      if (!satisfiesNodeVersionForMongo3and4 && !satisfiesNodeVersionForMongo5 && !satisfiesNodeVersionForMongo6) return
+
+      withVersions('express-mongo-sanitize', 'mquery', mqueryVersion => {
+        let mquery
+
+        const vulnerableMethodFilename = 'mquery-vulnerable-method.js'
+        let client, testCollection, tmpFilePath, dbName
+
+        before(() => {
+          return agent.load(['mongodb'], { client: false }, { flushInterval: 1 })
+        })
+
+        before(async () => {
+          const id = require('../../../../src/id')
+          dbName = id().toString()
+          const mongo = require(`../../../../../../versions/mongodb@${mongodbVersion}`).get()
+          mquery = require(`../../../../../../versions/mquery@${mqueryVersion}`).get()
+
+          client = new mongo.MongoClient(`mongodb://localhost:27017/${dbName}`, {
+            useNewUrlParser: true,
+            useUnifiedTopology: true
+          })
+          await client.connect()
+
+          testCollection = client.db().collection('Test')
+
+          const src = path.join(__dirname, 'resources', vulnerableMethodFilename)
+
+          tmpFilePath = path.join(os.tmpdir(), vulnerableMethodFilename)
+          try {
+            fs.unlinkSync(tmpFilePath)
+          } catch (e) {
+            // ignore the error
+          }
+          fs.copyFileSync(src, tmpFilePath)
+        })
+
+        after(async () => {
+          fs.unlinkSync(tmpFilePath)
+          await client.close()
+        })
+
+        prepareTestServerForIastInExpress('Test with mquery', expressVersion,
+          (testThatRequestHasVulnerability, testThatRequestHasNoVulnerability) => {
+            testThatRequestHasVulnerability({
+              fn: async (req, res) => {
+                await mquery()
+                  .find({
+                    name: req.query.key,
+                    value: [1, 2,
+                      'value',
+                      false, req.query.key]
+                  })
+                  .collection(testCollection)
+                  .then(() => {
+                    res.end()
+                  }).catch((err) => {
+                    res.end()
+                  })
+              },
+              vulnerability: 'NOSQL_MONGODB_INJECTION',
+              makeRequest: (done, config) => {
+                axios.get(`http://localhost:${config.port}/?key=value`).catch(done)
+              }
+            })
+
+            testThatRequestHasVulnerability({
+              testDescription: 'should have NOSQL_MONGODB_INJECTION vulnerability in correct file and line [find]',
+              fn: async (req, res) => {
+                const filter = {
+                  name: {
+                    child: [req.query.key]
+                  }
```

Comment:
why not just like in the `testThatRequestHasNoVulnerability` tests?
```suggestion
                  name: req.query.key
```
Here in the other tests